### PR TITLE
My account change password

### PR DESCRIPTION
**Pull request by daisyzarzo**
relates to https://github.com/bespoken/dashboard/issues/865

### CODE REVIEW

**Comment by daisyzarzo** [comment]
image

File: src/components/AuthForm/AuthForm.tsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class LoginGithub extends React.Component<LoginGithubProps, any> {
         padding: ".5rem",
     };
 
-    componentDidMount () {
+    componentDidMount() {
```
Comment (jperata) [inline src/components/AuthForm/AuthForm.tsx]: @AnkRaiza we have more than one componentDidMount in this one, shouldn't this cause some issues

File: src/components/MyAccount/Account.tsx [hunk 2]
```diff
@@ -22,20 +24,39 @@ export class Account extends React.Component<AccountProps, AccountState> {
...
+        if (auth.doTheUserHaveAPassword()) {
+            sectionProperty.push({
+                property: "Email: ",
+                propertyDescription: this.props.email,
+                href: " ",
+                hrefDescription: "Change password",
+                targetblank: false,
+                onclick: this.props.handleDialogToggle,
+            });
+            sectionProperty.push({
```
Comment (jperata) [inline src/components/MyAccount/Account.tsx]: just remove this from the if else condition
if (auth.doTheUserHaveAPassword) {
  section.push(changePasswordData);
}
section.push(deleteAccountData);

File: src/components/MyAccount/Password.test.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const handleDialogToggle = sinon.spy();
+
+describe("Password component", function () {
+    it("should render correctly", function () {
+        const wrapper = shallow(
+            <Password
+                handleCancelClick={handleDialogToggle} />
+        );
+        const buttons = wrapper.find(Button);
+        console.log(buttons);
```
Comment (jperata) [inline src/components/MyAccount/Password.test.tsx]: remove

File: src/components/MyAccount/Password.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            } else {
+                await this.setState({ ...this.state, error: error.message, success: "" });
+            }
+
+            return false;
+        }
+        return true;
+    }
+
+    handleChangeCurrentP = async (event: any) => {
```
Comment (jperata) [inline src/components/MyAccount/Password.tsx]: no reason to save letters, handleChangeCurrentPassword

File: src/components/MyAccount/Password.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        await this.setState({ ...this.state, currentPassword: event.target.value });
+        this.verifySavebutton();
+    }
+
+    handleChangeNewP = async (event: any) => {
```
Comment (jperata) [inline src/components/MyAccount/Password.tsx]: handleChangeNewPassword

File: src/components/MyAccount/Password.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        await this.setState({ ...this.state, newPassword: event.target.value });
+        this.verifySavebutton();
+    }
+
+    handleChangeRepeatP = async (event: any) => {
```
Comment (jperata) [inline src/components/MyAccount/Password.tsx]: handleChangeRepeatPassword

File: src/pages/MyAccountPage.tsx [hunk 7]
```diff
@@ -121,8 +128,26 @@ export class MyAccountPage extends React.Component<MyAccountPageProps, MyAccount
...
+        console.log("handledialog Toggle");
+        e && e.stopPropagation && e.stopPropagation();
+        this.setState((prevState) => ({
+            ...prevState,
+            changePasswordDialogActive: !prevState.changePasswordDialogActive,
+        }));
+    }
+
+    activeDialog = (e: any) => {
+        console.log("handledialog  activeDialog ");
```
Comment (jperata) [inline src/pages/MyAccountPage.tsx]: remove

File: src/pages/MyAccountPage.tsx [hunk 8]
```diff
@@ -121,8 +128,26 @@ export class MyAccountPage extends React.Component<MyAccountPageProps, MyAccount
             capturedLogs,
             monitoredSources,
             virtualDevicesAllowed
+
         });
     }
+    handleDialogToggle = (e: any) => {
+        console.log("handledialog Toggle");
```
Comment (jperata) [inline src/pages/MyAccountPage.tsx]: remove

File: src/pages/MyAccountPage.tsx [hunk 9]
```diff
@@ -121,8 +128,26 @@ export class MyAccountPage extends React.Component<MyAccountPageProps, MyAccount
...
+        e && e.stopPropagation && e.stopPropagation();
+        this.setState((prevState) => ({
+            ...prevState,
+            changePasswordDialogActive: !prevState.changePasswordDialogActive,
+        }));
+    }
+
+    activeDialog = (e: any) => {
+        console.log("handledialog  activeDialog ");
+        // e && e.stopPropagation && e.stopPropagation();
```
Comment (jperata) [inline src/pages/MyAccountPage.tsx]: readd

File: src/pages/MyAccountPage.tsx [hunk 10]
```diff
@@ -140,7 +165,19 @@ export class MyAccountPage extends React.Component<MyAccountPageProps, MyAccount
                     capturedLogs={this.state.capturedLogs}
                     monitoredSources={this.state.monitoredSources}
                     virtualDevicesAllowed={this.state.virtualDevicesAllowed}
+                    handleDialogToggle={this.activeDialog}
                 />
+                <Dialog
+                    // className={validationStyle.dialog}
```
Comment (jperata) [inline src/pages/MyAccountPage.tsx]: remove

File: src/services/auth.ts [hunk 11]
```diff
@@ -134,6 +134,27 @@ namespace auth {
         return Promise.all([matchProm, sizeProm]);
     }
 
+    export async function resetUserPassword(currentPassword: string, newPassword: string, repeatPassword: string,
+        auth: any = remoteservice.defaultService().auth()) {
+
+        let user: any = auth.currentUser;
```
Comment (jperata) [inline src/services/auth.ts]: const

File: src/services/auth.ts [hunk 12]
```diff
@@ -134,6 +134,27 @@ namespace auth {
...
+
+        return Promise.all([validatePassword(newPassword, repeatPassword),
+        user.reauthenticate(credential)]).then(function (result: any) {
+            return user.updatePassword(newPassword);
+        });
+
+    }
+
+    export const doTheUserHaveAPassword = (auth: remoteservice.auth.Auth = remoteservice.defaultService().auth()) => {
+        let user: any = auth.currentUser;
```
Comment (jperata) [inline src/services/auth.ts]: cont

**Comment by jperata** [comment]
Add screenshots for success and also screenshots for cases where the first password is incorrect, and where the passwords don't match.

File: src/components/MyAccount/Account.tsx (hunk 2 above)
Comment (daisyzarzo) [reply src/components/MyAccount/Account.tsx]: the logic is not that simple, this code is necesary

File: src/components/MyAccount/Password.tsx (hunk 4 above)
Comment (daisyzarzo) [reply src/components/MyAccount/Password.tsx]: it is necessary for disable button function

**Comment by daisyzarzo** [comment]
image
image


image

**Comment by dmarvp** [comment]
@daisyzarzo the modal is way too wide. Please reduce the size of it. Also, when changing the password correctly, the success message should be shown on a SnackBar after the modal has been closed.  (AC4)

**Comment by daisyzarzo** [comment]
image

**Comment by daisyzarzo** [comment]
image

**Comment by daisyzarzo** [comment]
image

**Comment by daisyzarzo** [comment]
image

**Pull request merged**